### PR TITLE
Support count for paragraph and sentence text objects

### PR DESCRIPTION
**Pull request by krobelus**
Part of #795

### CODE REVIEW

**Comment by lenormf** [comment]
Another way of implementing that is in #1602

File: src/selectors.cc [hunk 1]
```diff
@@ -494,58 +494,61 @@ select_sentence(const Context& context, const Selection& selection,
...
 
-    if (not (flags & ObjectFlags::ToEnd) and first != buffer.begin())
-    {
-        BufferIterator prev_non_blank = first-1;
-        skip_while_reverse(prev_non_blank, buffer.begin(),
-                           [](char c) { return is_horizontal_blank(c) or is_eol(c); });
-        if (is_end_of_sentence(*prev_non_blank))
-            first = prev_non_blank;
-    }
+    for (++count; count > 0; --count) {
```
Comment (mawww) [inline src/selectors.cc:499]: `{` on its own line please

File: src/selectors.cc [hunk 2]
```diff
@@ -494,58 +494,61 @@ select_sentence(const Context& context, const Selection& selection,
...
+        {
+            BufferIterator prev_non_blank = first-1;
+            skip_while_reverse(prev_non_blank, buffer.begin(),
+                               [](char c) { return is_horizontal_blank(c) or is_eol(c); });
+            if (is_end_of_sentence(*prev_non_blank))
+                first = prev_non_blank;
+        }
 
-    BufferIterator last = first;
+        if (last == BufferIterator{}) last = first;
```
Comment (mawww) [inline src/selectors.cc:509]: `last = first` on its own line

File: src/selectors.cc [hunk 3]
```diff
@@ -558,51 +561,53 @@ select_paragraph(const Context& context, const Selection& selection,
...
-    if (not (flags & ObjectFlags::ToEnd) and first.coord() > BufferCoord{0,1} and
-        is_eol(*(first-1)) and first-1 != buffer.begin() and is_eol(*(first-2)))
-        --first;
-    else if ((flags & ObjectFlags::ToEnd) and
-             first != buffer.begin() and (first+1) != buffer.end() and
-             is_eol(*(first-1)) and is_eol(*first))
-        ++first;
-
-    BufferIterator last = first;
+    for (++count; count > 0; --count) {
```
Comment (mawww) [inline src/selectors.cc:566]: same

**Comment by krobelus** [comment]
Done; I have added some tests

**Pull request merged**